frontend/src/modules/LoadTestContentRedux.js

```diff
@@ -14,24 +16,58 @@ const updateTestBackgroundState = testBackground => ({
 // API Calls
 const getTestMetaData = testName => {
   return async function() {
-    let metaDataContent = await fetch(`/api/test-meta-data/?test_name=${testName}`, {
-      method: "GET",
-      Accept: "application/json",
-      "Content-Type": "application/json",
-      cache: "default"
-    });
-    return await metaDataContent.json();
+    // test is private
+    if (!testName.public) {
```

Comment:
Instead of adding the auth token based on whether the test is public, you could do something like this, where you add it based on if the token exists:

```
return async function() {
    let headers = {
      Accept: "application/json",
      "Content-Type": "application/json",
      cache: "default"
    };
    if (AUTH_TOKEN) {
      headers.Authorization = "JWT " + AUTH_TOKEN;
    }
    let metaDataContent = await fetch(`/api/test-meta-data/?test_name=${testName}`, {
      method: "GET",
      headers: headers
    });
    return await metaDataContent.json();
  };
```

Reply:
Refactored the logic as suggested.
See commit #https://github.com/code-for-canada/project-thundercat/pull/461/commits/22e9c93d02fe5e2153bdf220da10e19d1c6c8ef6